### PR TITLE
.github: add act support for conformance-gke

### PR DESCRIPTION
**Pull request by aanm**
Running GH workflows locally is possible with the help of
nektos/act. The changes performed on the GKE workflow were necessary
to skip the steps that do not need to run locally. This could be
achieved with env.ACT as that environment variable is always set when
running act against a GH workflow.

Signed-off-by: André Martins <andre@cilium.io>

### CODE REVIEW

File: .github/workflows/conformance-gke.yaml [hunk 1]
```diff
@@ -152,6 +157,7 @@ jobs:
             --helm-set=hubble.relay.image.repository=quay.io/${{ github.repository_owner }}/hubble-relay-ci \
             --helm-set=hubble.relay.image.tag=${SHA} \
             --helm-set=agentNotReadyTaintKey=ignore-taint.cluster-autoscaler.kubernetes.io/cilium-agent-not-ready \
+            --helm-set=debug.enabled=true \
```
Comment (michi-covalent) [inline .github/workflows/conformance-gke.yaml:160]: i guess this is unrelated?

File: .github/workflows/conformance-gke.yaml [hunk 2]
```diff
@@ -383,7 +397,7 @@ jobs:
           target_url: ${{ env.check_url }}
 
       - name: Set commit status to cancelled
-        if: ${{ cancelled() }}
+        if: ${{ !env.ACT && cancelled() }}
```
Comment (christarazi) [inline .github/workflows/conformance-gke.yaml:398]: I feel like if we add new jobs here, it's easy for someone to forget to add this to the conditions, but not sure we can do anything about that. :shrug:

**Review by michi-covalent — APPROVED** [review]
🚢

File: Documentation/contributing/testing/workflows.rst [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+Once you have ``GITHUB_TOKEN`` exported and ``secrets-file.txt`` created, run
+the following command. As you can see, ``git rev-parse HEAD`` will be used as
+the default commit but you can choose any commit that has a docker image
+available:
+
+::
+
+    act -s GITHUB_TOKEN=${GITHUB_TOKEN} \
+        --secret-file=secrets-file.txt \
+        --env ACT_SHA=`git rev-parse HEAD` \
```
Comment (christarazi) [inline Documentation/contributing/testing/workflows.rst]: Nit: Should we recommend `$(...)` syntax instead? Shellcheck complains about the ` synxtax

File: .github/workflows/conformance-gke.yaml (hunk 1 above)
Comment (aanm) [reply .github/workflows/conformance-gke.yaml:160]: correct, that's why it's on a different commit

File: Documentation/contributing/testing/workflows.rst [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+consequent creation of GKE clusters. This can easily be achieved by adding
+``if: ${{ !env.ACT }}`` in the "Create GKE cluster" step:
+
+::
+
+           - name: Create GKE cluster
+    +        if: ${{ !env.ACT }}
+             run: |
+               gcloud container clusters create ${{ env.clusterName }} \
+
```
Comment (christarazi) [inline Documentation/contributing/testing/workflows.rst]: Should we just require `act` runs to assume a GKE environment so that mistaken invocations of `act` don't cause a whole bunch of clusters?

File: .github/workflows/conformance-gke.yaml (hunk 1 above)
Comment (michi-covalent) [reply .github/workflows/conformance-gke.yaml:160]: is it something specific to GKE, or do we want debug enabled across all the conformance tests?

File: Documentation/contributing/testing/workflows.rst (hunk 4 above)
Comment (christarazi) [reply Documentation/contributing/testing/workflows.rst]: Maybe @nbusseneau has some thoughts given that he had some script-driven workflow in mind that he's mentioned previously

File: .github/workflows/conformance-gke.yaml (hunk 1 above)
Comment (aanm) [reply .github/workflows/conformance-gke.yaml:160]: Probably all of them since it was helpful for me to understand what was going on with the current GKE failure.

File: Documentation/contributing/testing/workflows.rst [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+ - :ref:`gcp_project_id`
+ - :ref:`gcp_sa_key`
+
+
+Once you have ``GITHUB_TOKEN`` exported and ``secrets-file.txt`` created, run
+the following command. As you can see, ``git rev-parse HEAD`` will be used as
+the default commit but you can choose any commit that has a docker image
+available:
+
+::
```
Comment (qmonnet) [inline Documentation/contributing/testing/workflows.rst]:
```suggestion
.. code-block:: shell-session
```

File: Documentation/contributing/testing/workflows.rst [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+
+           - name: Create GKE cluster
+    +        if: ${{ !env.ACT }}
+             run: |
+               gcloud container clusters create ${{ env.clusterName }} \
+
+and for the cluster cleanup:
+
+::
```
Comment (qmonnet) [inline Documentation/contributing/testing/workflows.rst]:
```suggestion
.. code-block:: diff
```

File: Documentation/contributing/testing/workflows.rst [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+        issue_comment \
+        -W .github/workflows/conformance-gke.yaml
+
+**Tips**:
+
+When debugging a GKE flake it might be worth disabling the deletion and
+consequent creation of GKE clusters. This can easily be achieved by adding
+``if: ${{ !env.ACT }}`` in the "Create GKE cluster" step:
+
+::
```
Comment (qmonnet) [inline Documentation/contributing/testing/workflows.rst]:
```suggestion
.. code-block:: diff
```

File: Documentation/contributing/testing/workflows.rst (hunk 3 above)
Comment (qmonnet) [reply Documentation/contributing/testing/workflows.rst]: Here and for the `SECRET=` commands

File: Documentation/contributing/testing/workflows.rst [hunk 8]
```diff
@@ -0,0 +1,118 @@
...
+
+.. _gcp_project_id:
+
+GCP Project ID
+^^^^^^^^^^^^^^
+
+The ``GCP_PROJECT_ID`` secret should point to your GCP Project ID, store this
+value in a secrets file:
+
+::
```
Comment (qmonnet) [inline Documentation/contributing/testing/workflows.rst]:
```suggestion
.. code-block:: shell-session
```

File: Documentation/contributing/testing/workflows.rst [hunk 9]
```diff
@@ -0,0 +1,118 @@
...
+   3. Store the JSON (``serviceaccount.json``) file somewhere safe.
+
+3. Add the right permissions to this service account:
+
+   1. Go to `Google Cloud Console > IAM & Admin > IAM <https://console.cloud.google.com/iam-admin/iam>`_.
+   2. Edit the permissions for that service account and add the "Kubernetes Engine Admin" role to it.
+
+4. Store the ``serviceaccount.json`` in a secrets file:
+
+::
```
Comment (qmonnet) [inline Documentation/contributing/testing/workflows.rst]:
```suggestion
.. code-block:: shell-session
```

**Review by qmonnet — APPROVED** [review]
Minor formatting nits on the doc, looks all good to me otherwise :+1:

**Review by nbusseneau — COMMENTED** [review]
This is sound overall but I'm not so keen on merging it because of how the workflows are structured right now. As Chris noted, steps' conditions management is becoming even more complicated that it already is. One thing we could do to improve the situation is to more properly bundle/split steps so that ACT checks need only be done a few times in the workflows, but at the moment the workflows are not built like that.

Happy to merge anyway if we think that's best, this can always be done later. It just makes it kinda difficult to apply to all workflows, but they're already a mess anyway so...
